### PR TITLE
[Test]:Created writer_test,go in pkg/viaduct/pkg/packer/

### PR DESCRIPTION
**Pull request by vivekbisen04**
/kind test  

## What this PR does / why we need it  

This PR improves test coverage for the `./pkg/viaduct/pkg/packer/writer.go` package by introducing unit tests. It enhances validation by:  

- Ensuring correctness of the `writer.go` functionality  
- Covering various edge cases effectively  
- Improving overall test coverage from **0% to 88%**  

## Which issue(s) this PR fixes  

Part of [lfx-mentorship] Enhance KubeEdge testing coverage initiative **#6101**  

## Screenshots  
Screenshot from 2025-02-09 16-43-37

## Command to test  

- **File path:** `./pkg/viaduct/pkg/packer/writer.go`  
- **Test execution command:**  

  ```sh
  go test -coverprofile=coverage.out ./pkg/viaduct/pkg/packer/...
  go tool cover -html=coverage.out -o coverage.html

### CODE REVIEW

File: pkg/viaduct/pkg/packer/writer_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	if n != len(data) {
+		t.Errorf("expected %d bytes written, got %d", len(data), n)
+	}
+
+	// Writer with nil io.Writer
+	nilWriter := NewWriter(nil)
+	_, err = nilWriter.Write(data)
+	if err == nil {
+		t.Errorf("expected error, got nil")
```
Comment (wbc6080) [inline pkg/viaduct/pkg/packer/writer_test.go:59]:
```suggestion
		t.Error("expected error, got nil")
```

File: pkg/viaduct/pkg/packer/writer_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	failingWriter := NewWriter(&mockWriter{fail: true})
+	_, err = failingWriter.Write(data)
+	if err == nil {
+		t.Errorf("expected error, got nil")
```
Comment (wbc6080) [inline pkg/viaduct/pkg/packer/writer_test.go:65]:
```suggestion
		t.Error("expected error, got nil")
```

File: pkg/viaduct/pkg/packer/writer_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	_, err = nilWriter.Write(data)
+	if err == nil {
+		t.Error("expected error, got nil")
+	}
+
+	failingWriter := NewWriter(&mockWriter{fail: true})
+	_, err = failingWriter.Write(data)
+	if err == nil {
+		t.Error("expected error, got nil")
+	}
```
Comment (fujitatomoya) [inline pkg/viaduct/pkg/packer/writer_test.go:67]: i think we should check the expected error strings are stored.

```suggestion
	if err == nil {
		t.Error("expected error, got nil")
	} else if err.Error() != "mock write failure" {
		t.Errorf("unexpected error message: %v", err)
	}
```
